c/tskit/tables.c

```diff
@@ -7805,27 +7824,29 @@ ancestor_mapper_run(ancestor_mapper_t *self)
     tsk_id_t parent, current_parent;
     const tsk_edge_table_t *input_edges = &self->tables->edges;
     tsk_size_t num_edges = input_edges->num_rows;
+    const double *node_time = self->tables->nodes.time;
 
     if (num_edges > 0) {
         start = 0;
         current_parent = input_edges->parent[0];
-        for (j = 0; j < num_edges; j++) {
-            parent = input_edges->parent[j];
-            if (parent != current_parent) {
+        parent = input_edges->parent[0];
```

Comment:
Note this line isn't strictly necessary, but without it gcc now throws a warning about using `parent` uninitialised (probably because the initialisation is happening inside an if statement now)